roxie/roxiemem/roxiemem.hpp

```diff
@@ -417,6 +426,46 @@ interface IVariableRowHeap : extends IInterface
     virtual void *finalizeRow(void *final, memsize_t originalSize, memsize_t finalSize) = 0;
 };
 
+interface IBlockedRowReleaser : public IInterface
+{
+public:
+    virtual void releaseRows() = 0;
+};
+
+class CBlockedRowReleaser : public CInterfaceOf<IBlockedRowReleaser>
+{
+public:
+    CBlockedRowReleaser(unsigned _max, const void * * _rows) : num(0), max(_max), rows(_rows) {}
+    virtual ~CBlockedRowReleaser() {};
+
+    inline void appendRow(const void * row)
+    {
+        dbgassertex(num < max);
+        rows[num++] = row;
```

Comment:
Only that this should only be called from a sing thread.  I should make that clear